src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -116,12 +116,16 @@ public void Unregister(Workspace workspace, bool blockingShutdown = false)
             private async Task UnregisterAsync(Workspace workspace)
             {
                 Contract.ThrowIfFalse(workspace == _workspace);
-                Contract.ThrowIfNull(_analyzeTask);
 
-                _source.Cancel();
+                var analyzeTask = _analyzeTask;
+                if (analyzeTask != null)
+                {
+                    // wait for analyzer work to be finished
+                    await analyzeTask.ConfigureAwait(false);
```

Comment:
1. any concern about this task throwing, and thus us not getting to the next steps?
2. the old logic looked to eb "try to cancel the CTS, then await the task".  presumably so the task might respect the CTS.  The new logic always awaits the task, then cancels the source.  is the CTS relevant anymore if that's how things work?

Reply:
Good points.